File: Programmers/Lv1_모의고사.py
def solution(answers):
    one, two, three = [], [], []
    dicts = {1 : 0, 2 : 0, 3 : 0}

    for i in range(len(answers)):
        one.append(i%5+1)

        if i%2==0:
            two.append(2)
        elif i%8==5:
            two.append(4)
        elif i%8==7:
            two.append(5)
        else:
            two.append(i%8)

        if i%10<2:
            three.append(3)
        elif 2 <= i%10 < 6:
            three.append(i//2%5)
        elif 6 <= i%10 < 8:
            three.append(4)
        else:
            three.append(5)

    for i in range(len(answers)):
        if one[i] == answers[i]:
            dicts[1] += 1
        if two[i] == answers[i]:
            dicts[2] += 1
        if three[i] == answers[i]:
            dicts[3] += 1

    answer = [k for k, v in dicts.items() if v == max(list(dicts.values()))]
    
    return sorted(answer)

if __name__=='__main__':
    solution([1, 2, 3, 4, 5, 2, 4, 4, 5, 5, 3])

File: Programmers/C++/Level_1/Lv1_모의고사.cpp
#include <string>
#include <vector>

using namespace std;

int test1[5] = {1, 2, 3, 4, 5};
int test2[8] = {2, 1, 2, 3, 2, 4, 2, 5};
int test3[10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};

int max(int a, int b) {
    return a < b ? b : a;
}

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> score(3);
    int maxscore = 0;
    
    for (int i=0; i<answers.size(); i++) {
        if (test1[i%5] == answers[i])
            score[0]++;
        if (test2[i%8] == answers[i])
            score[1]++;
        if (test3[i%10] == answers[i])
            score[2]++;
    }
    
    maxscore = max(max(score[0], score[1]), score[2]);
    
    for (int i=0; i<3; i++) {
        if (score[i] == maxscore)
            answer.push_back(i+1);
    }
    return answer;
}